### DISCUSSION TITLE
"_up" folders generated while bundling

### DISCUSSION THREAD

**Discussion by enrico-calzavara** [parcel-bundler/parcel#8520] [Q&A] [answered]
Hi,

I have a project with the following structure:

```bash
assets/
├─ docs/
│  ├─ test.pdf
brands/
├─ foo/
│  ├─ index.html
src/
├─ main.js
```
`Index.html` contains an anchor pointing to test.pdf:

`<a target="_blank" href="../../assets/docs/test.pdf">Link</a>`

Once parcel is done running, in the `dist` folder I have the following:

```bash
dist/
├─ _up/
│  ├─ _up/
│  │  ├─ assets/
│  │  │  ├─ docs/
│  │  │  │  ├─ test.pdf
├─ index.(revision).js
├─ index.html
```
with `index.html` file containing: 

`<a target="_blank" href="/up_/up_/assets/docs/test.pdf">Link</a>`

When I open the generated `index.html`, everything is functioning properly.
However, I noticed the two `_up` folders in the generated structure and they looked a bit strange (I would have thought that the folders would have been flattened).
Does anyone know if it's correct that they're being generated in this way? Is this working as intended?

Thanks everyone

**Comment by mischnic** [comment, marked as answer]
It's working as intended, but maybe not as you intended 😬 

The "entry root" is `brands/foo` in your case, so `dist/index.html` will be created from `brands/foo/index.html`.
And `<a>` relative links are preserved just as they are relative on the filesystem.

So the pdf would have been put into `dist/../../assets/docs/test.pdf`, but we replace these `..` segments with `_up` to at least keep them inside the distdir.

What you'd really want in this case is the entry root pointing to the most toplevel folder, so that `dist/brands/foo/index.html` is created. But the entry root is not configurable at the moment and is always the nearest parent folder containing all entry points (= the files you specified in the CLI call `parcel build brands/foo/index.html`).

**Reply by enrico-calzavara** [reply]
Thank you @mischnic!
